File: app/src/main/java/mobi/porquenao/poc/kotlin/core/CalendarConverter.java
package mobi.porquenao.poc.kotlin.core;

import com.raizlabs.android.dbflow.converter.TypeConverter;

import java.util.Calendar;

@com.raizlabs.android.dbflow.annotation.TypeConverter
public class CalendarConverter extends TypeConverter<Long, Calendar> {

    @Override
    public Long getDBValue(Calendar model) {
        return model.getTimeInMillis();
    }

    @Override
    public Calendar getModelValue(Long data) {
        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(data);
        return calendar;
    }

}

File: app/src/main/java/mobi/porquenao/poc/kotlin/core/CalendarConverter.kt
package mobi.porquenao.poc.kotlin.core

import com.raizlabs.android.dbflow.converter.TypeConverter

import java.util.Calendar

@com.raizlabs.android.dbflow.annotation.TypeConverter
public class CalendarConverter : TypeConverter<Long, Calendar>() {

    override fun getDBValue(model: Calendar): Long? {
        return model.getTimeInMillis()
    }

    override fun getModelValue(data: Long?): Calendar {
        val calendar = Calendar.getInstance()
        calendar.setTimeInMillis(data!!)
        return calendar
    }

}
